### DISCUSSION TITLE
Opt out of complex key syntax

### DISCUSSION THREAD

**Discussion by bkilrain** [eemeli/yaml#282] [Q&A] [answered]
Hi there -

I've hit a behavior in the library that isn't wrong... but I'm not sure it's expected. Consider this snippet:
```
const doc = parseDocument('Key: {}');
doc.set('Key', null);
doc.toString() // => "? Key\n"
```

When there is a single key... and you set the key's value to null and then stringify, it results in the complex key syntax. My expectation is that we'd maintain the syntax of the original string (e.g. `"Key: null\n"`).

My expectations are met most of the time... I can only reproduce this behavior when:
* there is a single key in the document AND
* The value which is set to null is a collection (not scalar)

I've poked through the docs and can't find out if there is an option to opt out of the complex key syntax. Any ideas?

**Comment by eemeli** [comment, marked as answer]
The [ToString Option](https://eemeli.org/yaml/#tostring-options) you're looking for is `simpleKeys`.
